core/core_modules/update.js

```diff
@@ -1,128 +1,148 @@
 var git = require.once('../git.js'),
-	install = require('../install.js'),
-	/**
-	Default update period in milliseconds.
-	Once per day.
-	*/
-	defaultUpdatePeriod = 86400000,
-	timeout = null,
-	updatePeriod = defaultUpdatePeriod,
-	failedUpdateAttempts = 0,
-	currentSHA = null,
-	shutdown = null,
+    install = require('../install.js'),
+    /**
+    Default update period in milliseconds.
+    Once per day.
+    */
+    defaultUpdatePeriod = 86400000,
+    timeout = null,
+    updatePeriod = defaultUpdatePeriod,
+    failedUpdateAttempts = 0,
+    currentSHA = null,
+    shutdown = null,
 
-	checkForHash = function() {
-		git.getSHAOfRemoteMaster(function(err, consoleOutput) {
-			if (err) {
-				console.debug('Failed to get SHA hash of remote origin/master\n Error: ' + consoleOutput);
-				return false;
-			}
-			if (currentSHA !== consoleOutput) {
-				currentSHA = consoleOutput;
-				return true;
-			}
+    checkForHash = function() {
+        git.getSHAOfRemoteMaster(function(err, consoleOutput) {
+            if (err) {
+                console.debug('Failed to get SHA hash of remote origin/master\n Error: ' + consoleOutput);
+                return false;
+            }
+            if (currentSHA !== consoleOutput) {
+                currentSHA = consoleOutput;
+                return true;
+            }
 
-			return false;
-		});
-	},
+            return false;
+        });
+    },
 
-	setUpdateTimer = function() {
-		timeout = setTimeout(function() {
-			git.pull(function(err, consoleOutput) {
-				// Well something is wrong if the update process has failed for 3 periods.
-				// prevent restart and stop auto updating.
-				if (failedUpdateAttempts > 2) {
-					return;
-				}
+    setUpdateTimer = function() {
+        timeout = setTimeout(function() {
+            git.pull(function(err, consoleOutput) {
+                // Well something is wrong if the update process has failed for 3 periods.
+                // prevent restart and stop auto updating.
+                if (failedUpdateAttempts > 2) {
+                    return;
+                }
 
-				if (err) {
-					failedUpdateAttempts++;
-					console.debug('Periodic auto update failed. Manual intervention is probably required.\n Error: ' + consoleOutput);
-					setUpdateTimer();
-					return;
-				}
-                
-				try {
-					install.update();
-				}
-				catch (e) {
-					console.critical(e);
-					api.sendMessage('Periodic auto update of NPM packages failed. Manual NPM intervention will be required.', event.thread_id);
-				}
+                if (err) {
+                    failedUpdateAttempts++;
+                    console.critical(err);
+                    console.debug('Periodic auto update failed. Manual intervention is probably required.\n Error: ' + consoleOutput);
+                    setUpdateTimer();
+                }
+                else {
+                    git.submoduleUpdate(function(err) {
+                        if (err, consoleOutput) {
```

Comment:
@Awarua- You do know that this will have the same outcome as
```
if (consoleOutput) {
...
```
right? Which is not what we want cause even in the valid case there is console output.